tasks/RedHat.yml

```diff
@@ -11,20 +25,40 @@
     option: enabled
     value: 1
 
-- name: "Add elasticsearch repo"
+- name: "Add elasticsearch 1.7.x repo"
   yum_repository:
     name: "elasticsearch-1.7"
     description: "Elasticsearch repository for 1.7.x packages"
     file: elasticsearch_repo
     gpgkey: "http://packages.elastic.co/GPG-KEY-elasticsearch"
     gpgcheck: yes
     baseurl: "http://packages.elastic.co/elasticsearch/1.7/centos"
+  when: elasticsearch_version is version_compare('5.0', '<')
+
+- name: "Add elasticsearch 5.x repo"
+  yum_repository:
+    name: "elasticsearch-5.x"
+    description: "Elasticsearch repository for 5.x packages"
+    file: elasticsearch_repo
+    gpgkey: "https://artifacts.elastic.co/GPG-KEY-elasticsearch"
+    gpgcheck: yes
+    baseurl: "https://artifacts.elastic.co/packages/5.x/yum"
+  when: elasticsearch_version is version_compare('5.0', '>=') and elasticsearch_version is version_compare('6.0', '<')
```

Comment:
same than above, multiple "And" statements should go on a line on it's own to improve readability